mozci/data/sources/treeherder/__init__.py

```diff
@@ -78,21 +78,47 @@ def run_push_tasks(self, **kwargs):
                 .prefetch_related("jobnote_set")
             )
 
-            tasks = []
+            tasks = {}
 
             for task in self.normalize(jobs):
                 if task.get("tags"):
                     task["tags"] = {t["name"]: t["value"] for t in task["tags"]}
 
-                tasks.append(task)
+                tasks[task["id"]] = task
             return tasks
 
         logger.trace(
             "Unable to reach Treeherder as a datasource because the Job model "
             "was not available to import."
         )
-        return []
-
-    def run_push_tasks_tags(self, **kwargs):
-        # These are not needed for Push Health
         return {}
+
+    def run_push_tasks(self, branch, rev):
+        tasks = self._get_tasks(branch, rev)
+
+        return [
+            {
+                "id": task_data["id"],
+                "label": task_data["label"],
+                "tags": task_data["tags"],
```

Comment:
The only error I found so far is this line should be:

    "tags": task_data.get("tags", {}),

Because line 84 above shows `tags` may not exist in the dict.